lib/http.c

```diff
@@ -1456,8 +1456,10 @@ CURLcode Curl_http_done(struct connectdata *conn,
      data->state.negotiate.state == GSS_AUTHSENT) {
     /* add forbid re-use if http-code != 401/407 as a WA only needed for
      * 401/407 that signal auth failure (empty) otherwise state will be RECV
-     * with current code */
-    if((data->req.httpcode != 401) && (data->req.httpcode != 407))
+     * with current code.
+     * Do NOT close successful CONNECT-only connections. */
+    if((data->req.httpcode != 401) && (data->req.httpcode != 407) &&
+       (!data->set.connect_only || data->req.httpcode != 200))
```

Comment:
I'm a colleague of @tvbuehler 's and he's currently inactive on github, so I'll answer for him: yes, omitting the check for HTTP status code 200 works too. If I recall correctly, we added it as we thought only of our use case of extracting the socket after a successful HTTP CONNECT and this was the least invasive change for that use case.